utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp

```diff
@@ -901,33 +908,46 @@ int test(int test_case, int reader_type, const char *path, const char *outName,
         auto last_colot_temp = rocalGetIntValue(color_temp_adj);
         rocalUpdateIntParameter(last_colot_temp + 1, color_temp_adj);
 
-        rocalCopyToOutput(handle, mat_input.data, h * w * p);
+        RocalTensorList output_tensor_list = rocalGetOutputTensors(handle);
 
         std::vector<int> compression_params;
         compression_params.push_back(IMWRITE_PNG_COMPRESSION);
         compression_params.push_back(9);
 
-        mat_input.copyTo(mat_output(cv::Rect(col_counter * w, 0, w, h)));
-        std::string out_filename = std::string(outName) + ".png";   // in case the user specifies non png filename
-        if (display_all)
-            out_filename = std::string(outName) + std::to_string(index) + ".png";   // in case the user specifies non png filename
+        for(int idx = 0; idx < output_tensor_list->size(); idx++) { // Iterate over each output
+            auto output_tensor = output_tensor_list->at(idx);
+            int h = output_tensor->shape().at(1) * output_tensor->dims().at(0);
+            int w = output_tensor->shape().at(0);
+            if(first_run) { // Allocate cv matrix and output buffers for each output in first run and reuse in every iteration
+                mat_input.emplace_back(cv::Mat(h, w, cv_color_format));
+                mat_output.emplace_back(cv::Mat(h, w, cv_color_format));
+            }
+            
+            output_tensor->copy_data(mat_input[idx].data, ROCAL_MEMCPY_HOST);
+            mat_input[idx].copyTo(mat_output[idx](cv::Rect(0, 0, w, h)));
```

Comment:
There are 2 copies involved here, what does both of them do?
Add comments for better readability